File: codingTest/javascript/one/음계.js
const rl = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

function readLine() {
  return new Promise((reslove, reject) => {
    rl.on("line", (input) => {
      reslove(input);
      rl.close();
    });
  });
}

async function main() {
  var input = await readLine();
  var list = input.split(" ");
  //   list.forEach((e, i) => {
  //     list[i] = Number(e);
  //   });
  if (list.length != 8) return;
  var result;
  if (list[0] == 1) {
    result = "ascending";
    for (var i = 0; i < list.length - 1; i++) {
      if (list[i] > list[i + 1]) {
        result = "mixed";
        break;
      }
    }
  } else if (list[0] == 8) {
    result = "descending";
    for (var i = 0; i < list.length - 1; i++) {
      if (list[i] < list[i + 1]) {
        result = "mixed";
        break;
      }
    }
  } else result = "mixed";
  console.log(result);
}

main();

File: codingTest/kotlin/algorithm/src/one/음계.kt
package one

fun main() {
    val arr = readLine()!!.toString()
    val numArr = arr.split(' ').map { it.toInt() }
    val result = isMusic(numArr)
    println(result)
}

fun isMusic(arr: List<Int>): String {
    if (arr.size != 8) return "error"

    when (arr[0]) {
        1 -> {
            for (i in 0 until arr.size-1) if (arr[i] > arr[i+1]) return "mixed"
            return "ascending"
        }
        8 -> {
            for (i in 0 until arr.size-1) if (arr[i] < arr[i + 1]) return "mixed"
            return "descending"
        }
        else -> { return "mixed" }
    }
}